test/e2e/kogitoapp_test.go

```diff
@@ -58,47 +58,92 @@ func TestKogitoApp(t *testing.T) {
 	}
 	// more clusters can be added
 	t.Run("kogitoapp", func(t *testing.T) {
-		t.Run("Cluster1", KogitoAppCluster)
+		t.Run("QuarkusJvm", deployKogitoQuarkusJvmExample)
+		t.Run("QuarkusNative", deployKogitoQuarkusNativeExample)
 	})
 }
 
-func KogitoAppCluster(t *testing.T) {
+func deployKogitoQuarkusJvmExample(t *testing.T) {
 	t.Parallel()
+
 	// initialize framework
 	ctx := framework.NewTestCtx(t)
 	defer ctx.Cleanup()
-	// initialize cluster resources
-	err := ctx.InitializeClusterResources(&framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: retryInterval})
-	if err != nil {
-		log.Fatalf("failed to initialize cluster resources: %v", err)
-	}
-	log.Info("Initialized cluster resources")
-	// get our namespace
+	// get global framework variables
+	f := framework.Global
+
+	initializeKogitoOperator(t, f, ctx)
+
+	gitProjectURI := "https://github.com/kiegroup/kogito-examples"
+	contextDir := "drools-quarkus-example"
+	appName := "example-quarkus-jvm"
 	namespace, err := ctx.GetNamespace()
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalf("could not get namespace: %v", err)
 	}
-	// get global framework variables
-	f := framework.Global
-	// wait for kogito-operator to be ready
-	err = e2eutil.WaitForOperatorDeployment(t, f.KubeClient, namespace, "kogito-cloud-operator", 1, time.Second*20, time.Second*40)
+	kogitoService := &v1alpha1.KogitoApp{
```

Comment:
I guess this code is pretty similar for both use cases. Could you create separate functions for those?

Reply:
Yes, I put this code into separate file.